src/ChunkMap.cpp

```diff
@@ -1316,6 +1316,12 @@ void cChunkMap::DoExplosionAt(double a_ExplosionSize, double a_BlockX, double a_
 
 				double Impact = (1 - ((Length / ExplosionSizeInt) / 2)) * EntityExposure;
 
+				if (a_Entity.IsTNT() || a_Entity.IsFallingBlock())
```

Comment:
Shouldn't this be done by overriding `cFallingBlock::GetExplosionExposureRate` and `cTNTEntity::GetExplosionExposureRate`?

Reply:
Doesn't work well with the impact calculation in this case.

Reply:
In what way?

Reply:
@mathiascode bump?

Reply:
With a larger exposure rate, tnt and falling blocks either don't fly far enough, or fly way too far away. I don't have time to look deeper into how the algorithm works right now, feel free to mess around with these changes if you want.